File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
# Again, we pretend the codes don't match directly to list indices, and create a dict out of them, as if they were more complex

colours = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white"
]

colour_codes = dict(zip(colours, range(0, 10)))

def value(colors):
    # Concatenate codes of first two colors
    return int("".join([str(colour_codes[color]) for color in colors])[:2])

File: solutions/php/resistor-color-duo/1/ResistorColorDuo.php
<?php

declare(strict_types=1);

// first two digits only
// concat keys of first two

class ResistorColorDuo
{   
    public function getColorsValue(array $colors): int
    {   
        // pretty sure arrays are explicitly key:value hashtables in PHPland, where key is index
        $ALL_COLOURS = ["black", "brown","red","orange","yellow","green","blue","violet","grey","white"];
        // so grab keys, concatenate them, and return as int
        return +(array_search($colors[0], $ALL_COLOURS) . array_search($colors[1], $ALL_COLOURS));
    }
}
